code/game/objects/items/dehy_carp.dm

```diff
@@ -0,0 +1,46 @@
+/*
+ *	Dehydrated Carp
+ *	Instant carp, just add water
+ */
+
+// Child of carpplushie because this should do everything the toy does and more
+/obj/item/toy/carpplushie/dehy_carp
+	var/mob/owner = null	// Carp doesn't attack owner, set when using in hand
+	var/owned = 0	// Boolean, no owner to begin with
+
+// Attack self
+/obj/item/toy/carpplushie/dehy_carp/attack_self(mob/user)
+	src.add_fingerprint(user)	// Anyone can add their fingerprints to it with this
+	if(!owned)
+		to_chat(user, "<span class='notice'>[src] stares up at you with friendly eyes.</span>")
+		owner = user
+		owned = 1
+	return ..()
+
+
+/obj/item/toy/carpplushie/dehy_carp/afterattack(obj/O, mob/user, proximity)
+	if(!proximity)
+		return
+	if(istype(O,/obj/structure/sink))
+		to_chat(user, "<span class='notice'>You place [src] under a stream of water...</span>")
+		user.drop_item()
+		loc = get_turf(O)
+		return Swell()
+	..()
+
+/obj/item/toy/carpplushie/dehy_carp/proc/Swell()
+	desc = "It's growing!"
+	visible_message("<span class='notice'>[src] swells up!</span>")
+
+	// Animation
+	icon = 'icons/mob/animal.dmi'
+	flick("carp_swell", src)
+	// Wait for animation to end
+	sleep(6)
+	if(!src || qdeleted(src))//we got toasted while animating
+		return
+	// Make carp non-hostile to user, yes this means
+	owner.reagents.add_reagent(CARPPHEROMONES, 30)
+	//Make space carp
+	new /mob/living/simple_animal/hostile/carp(get_turf(src))
```

Comment:
Less health, less damage to mobs. Although I could make special carp and baby carp spawned by the dehydrated one that do more damage to mobs, but carp as they are can already break bones